tests/index_test.js

```diff
@@ -265,6 +265,58 @@ describe('rehydrate', () => {
     });
 });
 
+describe('StyleSheet.extend', () => {
+    beforeEach(() => {
+        global.document = jsdom.jsdom();
+        reset();
+    });
+
+    afterEach(() => {
+        global.document.close();
+        global.document = undefined;
+    });
+
+    it('uses a new selector handler', done => {
+        const descendantHandler = (selector, baseSelector,
```

Comment:
What would these arguments look like in this situation:

```javascript
{
  foo: {
    '^bar': {
      '^baz': {
      }
    }
  }
}
```

Just trying to think of edge cases to make sure that the plugin system makes sense.